app/onepassword_events_api/appserver/static/javascript/views/app_150.js

```diff
@@ -89,30 +94,42 @@ define(["react", "splunkjs/splunk"], function (react, splunk_js_sdk) {
       } catch (error) {
         return "Invalid JSON Web Token - " + error.message;
       }
-      if (!payload[aud] || payload[aud].length !== 1) {
+      if (!payload.aud || payload.aud.length !== 1) {
         return "Invalid JSON Web Token - missing aud";
       }
-      if (payload[aud][0] === audienceDEPRECATED) {
+      if (payload.aud[0] === audienceDEPRECATED) {
         return "Please generate a new token";
       }
     }
 
     render() {
       return e("div", null, [
         e(
-          "h2",
+          "h1",
           null,
-          "1Password Events API for Splunk Setup Page - Version 1.4.1"
+          "1Password Events API for Splunk Setup Page - Version 1.5.0"
         ),
+        this.state.apps.length > 0 &&
+          e("div", null, [
+            e("div", { class: "warning" }, [
+              e("h3", null, [
+                "WARNING: Any installed app could gain access to your token. Before saving it below, make sure you know and trust the following applications:",
+              ]),
+              e("div", null, this.state.apps.join(", ")),
```

Comment:
Hmm, I wonder if we should be listing the apps explicitly, or simply saying "Make sure you trust any installed apps" 🤔 I just want to be careful about walking the line between giving appropriate warning, and making it look like we're blaming other apps for being insecure.

Reply:
Yeah, thinking about it a bit more, I don't want us to list all the apps at this point, especially because it looks like it lists internal Splunk processes too:

![CleanShot 2021-07-21 at 15 47 51](https://user-images.githubusercontent.com/1508872/126550682-a281ecbd-b1d9-4271-b429-9b7aa8a126bc.png)

Also let's make sure we get the Words team's eyes on this copy.

Reply:
That's a good point and the "WARNING" might be a bit loud. However, the user is going to need to then leave this page and go to the installed apps settings in order to go through the list of installed apps to verify they trust each. It kind of disrupts the onboarding flow and they will also need to look through the splunk listed items there as well. If we are ok with that, then I'm happy to remove this.

Reply:
Yeah, I'm fine with that. 👍 

And I agree, the WARNING might be a bit loud, I'll leave that up to the Words team.